pkg/generate/app/dockerimagelookup.go

```diff
@@ -75,7 +75,7 @@ func (r DockerClientSearcher) Search(precise bool, terms ...string) (ComponentMa
 		termMatches := ScoredComponentMatches{}
 
 		// first look for the image in the remote docker registry
-		if r.RegistrySearcher != nil {
+		if r.RegistrySearcher != nil && ref.Name != "*" {
```

Comment:
OK. I will use `ref.String()`. I would like to emphasize the unacceptability of such values in `ref`.